bliss/models/decoder.py

```diff
@@ -349,6 +246,62 @@ def _validate_border_padding(self, border_padding):
         return int(border_padding)
 
 
+def reconstruct_image_from_ptiles(
+    image_ptiles: Tensor, tile_slen: int, border_padding: int
+) -> Tensor:
+    """Reconstruct an image from a tensor of padded tiles.
+
+    Given a tensor of padded tiles and the size of the original tiles, this function
+    combines them into a full image with overlap.
+
+    For now, the reconstructed image is assumed to be square. However, this function
+    can easily be refactored to allow for different numbers of horizontal or vertical
+    tiles.
+
+    Args:
+        image_ptiles: Tensor of size
+            (batch_size x n_tiles_per_image x n_bands x ptile_slen x ptile_slen)
+        tile_slen:
+            Size of the original (non-overlapping) tiles.
+        border_padding:
+            Amount of border padding to keep beyond the original tiles.
+
+    Returns:
+        Reconstructed image of size (batch_size x n_bands x height x width)
+    """
+    _, n_tiles_per_image, _, ptile_slen, _ = image_ptiles.shape
+    n_tiles_width = np.sqrt(n_tiles_per_image)
+    # check it is an integer
+    assert n_tiles_width % 1 == 0
+    n_tiles_width = int(n_tiles_width)
+    n_tiles_height = n_tiles_width
+
+    image_ptiles_prefold = rearrange(image_ptiles, "b n c h w -> b (c h w) n")
+    kernel_size = (ptile_slen, ptile_slen)
+    stride = (tile_slen, tile_slen)
+    ntiles_hw = (n_tiles_height, n_tiles_width)
+    output_size = calc_output_size(kernel_size, stride, ntiles_hw)
+
+    folded_image = F.fold(image_ptiles_prefold, output_size, kernel_size, stride=stride)
+
+    # In default settings of ImageDecoder, no borders are cropped from
+    # output image. However, we may want to crop
+    max_padding = (ptile_slen - tile_slen) / 2
+    assert max_padding % 1 == 0
+    max_padding = int(max_padding)
+    crop_idx = max_padding - border_padding
+    return folded_image[:, :, crop_idx : (-crop_idx or None), crop_idx : (-crop_idx or None)]
```

Comment:
Why do you need the `or None` ? Is that in case crop_idx = 0 ?

Reply:
Yes, the "or None" will kick in when crop_idx=0, which means we don't want to crop anything.